google-cloud-bigtable/acceptance/bigtable/table/mutate_row_test.rb

```diff
@@ -74,6 +74,88 @@
     row.cells[family].select{|v| v.qualifier == qualifier}.length.must_equal 1
   end
 
+  it "raises when set_cell with incorrect Time timestamp" do
+    postfix = random_str
+    row_key = "setcell-#{postfix}"
+    qualifier = "mutate-row-#{postfix}"
+
+    table.granularity.must_equal :MILLIS
+
+    # Set cell
+    entry = table.new_mutation_entry(row_key)
+    err = expect do
+      entry.set_cell(
+        family, qualifier, "mutatetest value #{postfix}", timestamp: Time.now
+      )
+    end.must_raise Google::Protobuf::TypeError
+    err.message.must_match "Expected number type for integral field 'timestamp_micros' (given Time)."
+  end
+
+  it "raises when set_cell with incorrect microsecond integer timestamp" do
+    postfix = random_str
+    row_key = "setcell-#{postfix}"
+    qualifier = "mutate-row-#{postfix}"
+    timestamp_micros = (Time.now.to_f * 1000000).round
+    timestamp_micros += 1 if (timestamp_micros % 10).zero?
+
+    table.granularity.must_equal :MILLIS
+
+    # Set cell
+    entry = table.new_mutation_entry(row_key)
+    entry.set_cell(
+      family, qualifier, "mutatetest value #{postfix}", timestamp: timestamp_micros
+    )
+    err = expect do
+      table.mutate_row(entry)
+    end.must_raise Google::Gax::RetryError
+    err.message.must_match /Timestamp granularity mismatch. Expected a multiple of 1000 \(millisecond granularity\), but got #{timestamp_micros}/
+  end
+
+  it "raises when set_cell with incorrect millisecond integer timestamp" do
+    postfix = random_str
+    row_key = "setcell-#{postfix}"
+    qualifier = "mutate-row-#{postfix}"
+    timestamp_millis = (Time.now.to_f * 1000).to_i
```

Comment:
I think this test also has a 1 in 1000 chance of not raising.

```suggestion
    timestamp_millis = (Time.now.to_f * 1000).round
    timestamp_millis += 1 if (timestamp_millis % 10).zero?
```